lib/deepcrawl.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+const BPromise = require('bluebird'),
+  schema = require('../schemas/v2.0.0'),
+  needle = BPromise.promisifyAll(require('needle')),
+  humps = require('humps'),
+  _ = require('lodash');
+
+class DeepCrawl {
+  constructor(cfg) {
+    cfg = cfg || {};
+
+    this.apiId = cfg.apiId;
+    this.apiKey = cfg.apiKey;
+    this.getSessionToken = cfg.getSessionToken;
+    this.accountId = cfg.accountId;
```

Comment:
i'm not throwing, because none of them are technically required at this point.  i check for sessionToken on each method call, if it doesn't exist and getSessionToken also doesn't exist, then i throw an error.  same with accountId.  they can pass that in to each call as well.  apiId and apiKey aren't required right now.  they honestly don't have to be here.